src/libstd/sys/windows/stdio_uwp.rs

```diff
@@ -0,0 +1,90 @@
+#![unstable(issue = "0", feature = "windows_stdio")]
+
+use crate::io;
+use crate::sys::c;
+use crate::sys::handle::Handle;
+
+pub struct Stdin {
+    #[allow(dead_code)]
+    surrogate: u16,
+}
+pub struct Stdout;
+pub struct Stderr;
+
+const MAX_BUFFER_SIZE: usize = 8192;
+pub const STDIN_BUF_SIZE: usize = MAX_BUFFER_SIZE / 2 * 3;
+
+pub fn get_handle(handle_id: c::DWORD) -> io::Result<c::HANDLE> {
+    let handle = unsafe { c::GetStdHandle(handle_id) };
+    if handle == c::INVALID_HANDLE_VALUE {
+        Err(io::Error::last_os_error())
+    } else if handle.is_null() {
+        Err(io::Error::from_raw_os_error(c::ERROR_INVALID_HANDLE as i32))
+    } else {
+        Ok(handle)
+    }
+}
+
+fn write(handle_id: c::DWORD, data: &[u8]) -> io::Result<usize> {
+    let handle = get_handle(handle_id)?;
+    let handle = Handle::new(handle);
+    let ret = handle.write(data);
+    handle.into_raw(); // Don't close the handle
```

Comment:
FWIW I've been using `ManuallyDrop` for this recently and I think it works pretty well:

```rust
ManuallyDrop::new(handle).write(data)
```

(no need for local `let` or `into_raw` and such)

Reply:
I can integrate the simplification in this MR but it seems unrelated, so up to you!

Reply:
I think it's fine to add here since this is all code being added in this PR.

Reply:
This should now be fixed, hopefully that's what you had in mind!